app/src/main/java/com/trianguloy/urlchecker/modules/list/PatternModule.java

```diff
@@ -225,4 +228,100 @@ public Message(String pattern) {
             this.pattern = pattern;
         }
     }
+
+    /**
+     * On Android 10 and under, optional groups may yield a "null" in the replacement output instead of an empty string.
+     * Therefore, we just copy the implementation from newer versions
+     * https://github.com/TrianguloY/UrlChecker/issues/237
+     */
+    private static String replaceAll(String text, Pattern pattern, String replacement) {
```

Comment:
Is it possible to place an early-return-if here to call the original unmodified replaceAll if we are on an android version where the fix is already native?

Reply:
Sure